core/baseline-core-v1.0-psd01.md

```diff
@@ -2903,27 +2954,43 @@ BPI Edge Storage MUST ensure eventual consistency between edge storage and BPI u
+
 #### **[R304]**	
 Replication conflicts MUST be automatically detectable. 
 
+[[R304]](#r304) Testability: An example of external BPI storage that can detect replication conflicts automatically is the open-source [PostgreSQL](https://www.postgresql.org/) data base, and since PostgreSQL has an extensive test suite including replication tests, the requirement is testable.
+
 #### **[R305]** 
 Replication conflicts MUST be resolvable either automatically or manually.
 
+[[R305]](#r305) Testability: An example of external BPI storage that can resolve replication conflicts automatically (e.g. rejecting conflicting updates), as well as manually (e.g. update a table update SQL procedure) is the open-source [PostgreSQL](https://www.postgresql.org/) data base, and since PostgreSQL has an extensive test suite including replication tests, the requirement is testable.
+
 #### **[R306]**	
 BPI Edge Storage MUST use a secure and privacy-preserving wire protocol.
 
+[[R306]](#r306) Testability: libp2p as a very commonly used P2P message/wire protocol supports end-to-end encryption, see [here](https://github.com/libp2p/specs/blob/master/noise/README.md) for the NOISE specification and relevant tests [here](https://github.com/libp2p/go-libp2p/blob/master/test-plans/cmd/ping/main.go) as well as authentication and authorization solutions such as OAuth-2.0. This shows that the requirement is testable.
+
 #### **[D40]**	
 BPI Edge Storage SHOULD be able to cryptographically sign messages.
 
+[[D40]](#d40) Testability: An example of external BPI storage that can sign messages is the open-source [PostgreSQL](https://www.postgresql.org/) data base using SSL, and since PostgreSQL has an extensive test suite including SSL certificate testing, the requirement is testable.
+
 #### **[D41]**	
 BPI Edge Storage SHOULD be discoverable by BPI Workgroup members or their delegates within a BPI.
 
+[[D41]](#d41) Testability: A BPI Storage access endpoint can be listed in the `Services` section of a DID document. Since BPI Subjects as members of a Workgroup can be associated with a DID, and since a DID can be resolved to its DID Document including its 'Services' section, and since the W3C DID Standard has an existing test suite for DID methods claiming conformance, the requirement is testable.
+
 #### **[R307]**	
 BPI Edge Storage MUST support BPI identifiers and identity as defined in this document. See section [3 Identifiers, Identity and Credential Management](#3-identifiers-identity-and-credential-management).
 
+[[R307]](#r307) Testability: Since OAuth-2.0 can utilize DIDs which can be used as BPI identifiers and identities, and since the open-source [PostgreSQL](https://www.postgresql.org/) data base supports OAuth-2.0, and since PostgreSQL has an extensive test suite including OAuth-2.0 tests, the requirement is testable.
```

Comment:
Does the PostgreSQL test suite specifically cover the use of DIDs in OAuth-2? Otherwise the assertion has a gap in its justification

Reply:
@chaals it does not because since Oathu2 with DID is testable and Oauth2 is testable within postgres the entirety is testable. It does not mean that such tests exist yet (need to modify the postgres tests according to the Oauth2 tests with DIDs). Just that they are possible to create.

Reply:
I think you mean "the assertion doesn't have a gap in its justification", by your answer "it does not".

Since the text doesn't point out that the use of DIDs in OAuth 2 is testable, I think there is a logical gap there.

(Under the alternative interpretation of your comment, that there are not tests for OAuth using DID in PostgreSQL I still think there is a gap - suggesting that specific tests can be created just because there are already extensive tests is pretty weak logic, and the claims of testability are all about logic).

Reply:
will add "testably utilize" for DIDs and Oauth2 and a link to a reference with test.